barretenberg/cpp/src/barretenberg/execution_trace/execution_trace.cpp

```diff
@@ -55,12 +55,66 @@ template <class Flavor>
+        for (auto& wire : proving_key.polynomials.get_wires()) {
+            wire = Polynomial::shiftable(proving_key.circuit_size);
+        }
+        // Define selectors over the block they are isolated to
+        uint32_t offset = Flavor::has_zero_row ? 1 : 0;
+        if constexpr (IsGoblinFlavor<Flavor>) {
+            offset += builder.blocks.ecc_op.get_fixed_size(is_structured);
+            info("offset after ecc_op block is ", offset);
+        }
+        offset += builder.blocks.pub_inputs.get_fixed_size(is_structured);
+        info("offset after pub_inputs block is ", offset);
+        proving_key.polynomials.q_arith =
+            Polynomial(proving_key.circuit_size - offset, proving_key.circuit_size, offset);
+        offset += builder.blocks.arithmetic.get_fixed_size(is_structured);
+        info("offset after arithmetic block is ", offset);
+        proving_key.polynomials.q_delta_range =
+            Polynomial(builder.blocks.delta_range.size(), proving_key.circuit_size, offset);
+        offset += builder.blocks.delta_range.get_fixed_size(is_structured);
+        info("offset after delta_range block is ", offset);
+        proving_key.polynomials.q_elliptic =
+            Polynomial(builder.blocks.elliptic.size(), proving_key.circuit_size, offset);
+        offset += builder.blocks.elliptic.get_fixed_size(is_structured);
+        info("offset after elliptic block is ", offset);
+        proving_key.polynomials.q_aux = Polynomial(builder.blocks.aux.size(), proving_key.circuit_size, offset);
+        offset += builder.blocks.aux.get_fixed_size(is_structured);
+        info("offset after aux block is ", offset);
+        proving_key.polynomials.q_lookup = Polynomial(builder.blocks.lookup.size(), proving_key.circuit_size, offset);
+        offset += builder.blocks.lookup.get_fixed_size(is_structured);
+        info("offset after lookup block is ", offset);
+        if constexpr (HasDataBus<Flavor>) {
+            proving_key.polynomials.q_busread =
+                Polynomial(builder.blocks.busread.size(), proving_key.circuit_size, offset);
+            offset += builder.blocks.busread.get_fixed_size(is_structured);
+            info("offset after busread block is ", offset);
+        }
+        proving_key.polynomials.q_poseidon2_external =
+            Polynomial(builder.blocks.poseidon2_external.size(), proving_key.circuit_size, offset);
+        offset += builder.blocks.poseidon2_external.get_fixed_size(is_structured);
+        info("offset after poseidon2_external block is ", offset);
+        proving_key.polynomials.q_poseidon2_internal =
+            Polynomial(builder.blocks.poseidon2_internal.size(), proving_key.circuit_size, offset);
+        offset += builder.blocks.poseidon2_internal.get_fixed_size(is_structured);
+        info("offset after poseidon2_internal block is ", offset);
+        // set the other selector polynomials to full size
+        proving_key.polynomials.q_m = Polynomial(proving_key.circuit_size);
```

Comment:
could create get_non_gate_selectors() to just create all of those together in a loop.